cumulusci/tasks/salesforce/promote_2gp_package.py

```diff
@@ -0,0 +1,308 @@
+                )
+            raise  # pragma: no cover
+
+    def _query_SubscriberPackage(self, sp_id: str) -> Optional[Dict]:
+        """Queries for a SubscriberPackage with the given SubscriberPackageId"""
+        return self._query_one_tooling(
+            ["Id", "Name"],
+            "SubscriberPackage",
+            where_clause=f"Id='{sp_id}'",
+            raise_error=True,
+        )
+
+    def _query_SubscriberPackageVersion(self, spv_id: str) -> Optional[Dict]:
+        """Queries for a SubscriberPackageVersion record with the given SubscriberPackageVersionId"""
+        return self._query_one_tooling(
+            ["Id", "Dependencies", "ReleaseState", "SubscriberPackageId"],
+            "SubscriberPackageVersion",
+            where_clause=f"Id='{spv_id}'",
+            raise_error=True,
+        )
+
+    def _query_one_tooling(
+        self,
+        fields: List[str],
+        obj_name: str,
+        where_clause: str = None,
+        raise_error=False,
+    ) -> Optional[Dict]:
+        """
+        Queries the Tooling API and returns a _single_ sObject (or None).
+        See docstring of _query_tooling() for param info.
+        """
+        records = self._query_tooling(
+            fields, obj_name, where_clause=where_clause, raise_error=raise_error
+        )
+        return records[0] if records else None
+
+    def _query_tooling(
+        self,
+        fields: List[str],
+        obj_name: str,
+        where_clause: str = None,
+        raise_error: bool = False,
+    ) -> Optional[List[Dict]]:
+        """
+        Queires the Tooling API
+
+        @param fields: list of fields to query for
+        @param obj_name: name of the tooling API sObject you want to query
+        @param where_clause: the where clause for the query (everything _after_ 'WHERE')
+        @param return_one: If true, returns a single sObject record
+        @param raise_error: If True, raises an error if no records are found
+
+        @return: None if no records are found, single sObject if `return_one` is True,
+        or a list of sObject records if `return_one` is False.
+        """
+        query = f"SELECT {', '.join(fields)} FROM {obj_name}"
+        if where_clause:
+            query += f" WHERE {where_clause}"
+
+        res = self.tooling.query(query)
+
+        if not res["records"] or res["size"] == 0:
+            if raise_error:
+                raise CumulusCIException(f"No records returned for query: {query}")
+            return None
+
+        return res["records"]
+
+    def _filter_one_gp_deps(self, dependencies: List[Dict]) -> List[Dict]:
+        """Return only 1GP dependencies"""
+        return [d for d in dependencies if not d["is_2gp"]]
+
+    def _filter_two_gp_deps(self, dependencies: List[Dict]) -> List[Dict]:
+        """Return only 2GP dependencies"""
+        return [d for d in dependencies if d["is_2gp"]]
```

Comment:
Maybe _get_dependency_info should put the dependencies into the appropriate lists as it collects the info and return the multiple lists, instead of needing to filter later? Your call.